### PR TITLE
Compile with Lua 5.3, and fix memory corruption.

### PR DESCRIPTION
**Pull request by rohit-n**

### CODE REVIEW

**Comment by phobos2077** [comment]
Will it still work with Lua 5.1? We need full Luajit compatibility.

**Comment by alexeevdv** [comment]
I don't see anything specific to Lua 5.3 in this commit

**Comment by phobos2077** [comment]
I just wanted to point out that Luajit compatibility was one of the priorities (one of the reasons we choose Lua language to begin with). Need to keep it :)

**Comment by alexeevdv** [comment]
https://github.com/falltergeist/falltergeist/blob/master/CMakeLists.txt#L60

File: src/Lua/Script.cpp [hunk 1]
```diff
@@ -151,8 +151,9 @@ bool Script::get(const std::string& name, bool defaultValue)
     {
         return defaultValue;
     }
-
-    return lua_toboolean(_lua_State, -1);
+    bool ret = lua_toboolean(_lua_State, -1);
```
Comment (dreamer-dead) [inline src/Lua/Script.cpp]: Can we generalize those lines?
Somethings like 

```
#include "LuaBridge/detail/Stack.h"

template <typename T>
T get(lua_State* state) {
    struct StackGuard {
        ~StackGuard() { lua_pop(_state, 1); }

        lua_State* const _state;
    } const guard {state};

    return Stack<T>::get(state);
}

...

return get<bool>(_lua_State);
```
Stack helper from LuaBridge uses luaL_check{number|string} but we already checked type of the stack value and thus we don't need to use lua_to{} API, IMHO

**Comment by isage** [comment]
Why do we even need to manage stack by hand?
>When a C function returns, Lua clears its stack

**Comment by rohit-n** [comment]
@isage I'm new to Lua myself, but not popping the stack was causing an immediate crash on Linux. Valgrind was reporting invalid reads at the last get() in Settings.cpp, and an invalid write when closing _lua_State.

**Comment by isage** [comment]
Can you please provide full stacktrace? Is it only with lua 5.3 or with 5.1 too?

**Comment by isage** [comment]
Oh, sorry, i've missed, that we just reading values from lua script. Yep, we should cleanup stack after we modified it with lua_getglobal.

**Pull request merged**